app/controllers/api/enterprise_fees_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Api
+  class EnterpriseFeesController < BaseController
+    respond_to :json
+
+    def destroy
+      authorize! :destroy, @enterprise_fee
+
+      if enterprise_fee.destroy
+        render text: I18n.t(:successfully_removed), status: 204
+      else
+        render json: { errors: enterprise_fee.errors }, status: 403
```

Comment:
Yes, my problem is that we are calling the API so the response should be in JSON right ? Also, if I render text it'll be with something like `render text: @enterprise_fee.errors[:base], status: 403` which is not ideal for other errors that might occur. I'm looking for a way to render just the error's full message via Angular, so that I can keep rendering JSON.

Reply:
@mkllnk @Matt-Yorkley any ideas on this one ? Should I render text for now and open an issue to remember it ?

Reply:
You are right that the API should return JSON, but that can happen in a different pull request. For now, it's best to keep the current API (text) which is compatible with the used Spree code: https://github.com/openfoodfoundation/spree/blob/69db1c090f3711088d84b524f1b94d25e6d21616/backend/app/assets/javascripts/admin/admin.js.erb#L174-L196

We don't want do rely on Spree's backend code and we want to get away from jQuery as well. So it would be another task to implement an Angular directive to do the same thing, but with JSON instead of text.